### ISSUE TITLE
Editing config.toml file saved content as JSON

### ISSUE THREAD

**Issue by ocdtrekkie** [hacdias/caddy-filemanager#77]
### 1. Which revision of File Manager are you running (check the footer on any page)?
Revision 02e2bec.

### 2. What are you trying to do?
I edited the title of the config.toml file and saved it. I discovered that the content in the config.toml file was now in JSON format.

### 3. What is your entire Caddyfile?
I deleted it already. :/ But it merely contained the baseURL, languageCode, and title fields in a config.toml file... formatted in JSON.

### 4. What did you expect to see?
I expected when editing config.toml, the format would remain correct for a toml file.

### 5. What did you see instead (give full error messages and/or log)?
Hugo was unable to parse the file because it didn't recognize the JSON in the TOML file.

### 6. How can someone who is starting from scratch reproduce this behavior as minimally as possible?
So I was asked to file this issue, it's kind of confusing to explain though. I am testing a [Hugo package for Sandstorm.io](https://apps.sandstorm.io/app/ksyud8q9h7rx9u001jn38216xe75a200avh7tg267wss6pq9zfkh?experimental=true) which includes Caddy and File Manager in it. The app package was developed by a developer who requires accessibility, and Caddy isn't particularly accessible, and I am testing it. This behavior was totally unexpected, and I am new to using Hugo, Caddy, FileManager, etc.

If I can get any help tracking down where this came from, I'd appreciate it.

**Comment by ndarilek** [comment]
FWIW, your entire Caddyfile is:

```
127.0.0.1:8001
errors stderr
log stdout
hugo /var/git {
  allow_commands false
  after_save "/opt/app/commit-on-save {path}"
}
```

**Comment by hacdias** [comment]
Hello @ocdtrekkie! Sorry for hearing this 😕 Have you got any error on DevTools or in Caddy's log?

This has happened before, but I thought it was already fixed. Could you try building Caddy from source with the newest version of File Manager, please?

**Comment by ocdtrekkie** [comment]
@hacdias That question would have to end up falling to @ndarilek in terms of building from source and such.

It looks like we're pushing Caddy log out to stdout, which I do have a capture on. As you can see at the end, after I edit the config.toml file again, I get a parsing error for "unexpected token":

```
GET / 200 539.601 ms - 1637
GET /stylesheets/style.css 200 8.170 ms - 111
GET /publicId 200 249.374 ms - 631
127.0.0.1 - [19/Mar/2017:02:09:52 +0000] "GET /admin/ HTTP/1.1" 200 12699
GET /admin/ 200 101.740 ms - -
127.0.0.1 - [19/Mar/2017:02:09:52 +0000] "GET /admin/_filemanagerinternal/css/normalize.css HTTP/1.1" 200 8053
GET /admin/_filemanagerinternal/css/normalize.css 200 5.443 ms - -
127.0.0.1 - [19/Mar/2017:02:09:52 +0000] "GET /admin/_filemanagerinternal/css/fonts.css HTTP/1.1" 200 4215
GET /admin/_filemanagerinternal/css/fonts.css 200 6.894 ms - -
127.0.0.1 - [19/Mar/2017:02:09:52 +0000] "GET /admin/_filemanagerinternal/css/styles.css HTTP/1.1" 200 18865
GET /admin/_filemanagerinternal/css/styles.css 200 4.946 ms - -
127.0.0.1 - [19/Mar/2017:02:09:52 +0000] "GET /admin/_filemanagerinternal/js/common.js HTTP/1.1" 200 14429
GET /admin/_filemanagerinternal/js/common.js 200 3.437 ms - -
127.0.0.1 - [19/Mar/2017:02:09:52 +0000] "GET /admin/_filemanagerinternal/js/listing.js HTTP/1.1" 200 15189
GET /admin/_filemanagerinternal/js/listing.js 200 7.912 ms - -
127.0.0.1 - [19/Mar/2017:02:09:52 +0000] "GET /admin/_hugointernal/js/application.js HTTP/1.1" 200 4429
GET /admin/_hugointernal/js/application.js 200 8.161 ms - -
127.0.0.1 - [19/Mar/2017:02:09:52 +0000] "GET /admin/_filemanagerinternal/css/roboto/normal-cyrillic.woff2 HTTP/1.1" 200 9996
GET /admin/_filemanagerinternal/css/roboto/normal-cyrillic.woff2 200 41.610 ms - -
127.0.0.1 - [19/Mar/2017:02:09:52 +0000] "GET /admin/_filemanagerinternal/css/roboto/normal-greek.woff2 HTTP/1.1" 200 7208
127.0.0.1 - [19/Mar/2017:02:09:52 +0000] "GET /admin/_filemanagerinternal/css/roboto/normal-greek-ext.woff2 HTTP/1.1" 200 1432
127.0.0.1 - [19/Mar/2017:02:09:52 +0000] "GET /admin/_filemanagerinternal/css/roboto/normal-vietnamese.woff2 HTTP/1.1" 200 4896
GET /admin/_filemanagerinternal/css/roboto/normal-greek.woff2 200 9.384 ms - -
GET /admin/_filemanagerinternal/css/roboto/normal-greek-ext.woff2 200 46.677 ms - 1432
GET /admin/_filemanagerinternal/css/roboto/normal-vietnamese.woff2 200 8.103 ms - -
127.0.0.1 - [19/Mar/2017:02:09:52 +0000] "GET /admin/_filemanagerinternal/css/roboto/normal-cyrillic-ext.woff2 HTTP/1.1" 200 15832
GET /admin/_filemanagerinternal/css/roboto/normal-cyrillic-ext.woff2 200 50.307 ms - -
127.0.0.1 - [19/Mar/2017:02:09:52 +0000] "GET /admin/_filemanagerinternal/css/roboto/normal-latin-ext.woff2 HTTP/1.1" 200 12008
GET /admin/_filemanagerinternal/css/roboto/normal-latin-ext.woff2 200 8.243 ms - -
127.0.0.1 - [19/Mar/2017:02:09:52 +0000] "GET /admin/_filemanagerinternal/css/roboto/normal-latin.woff2 HTTP/1.1" 200 14584
GET /admin/_filemanagerinternal/css/roboto/normal-latin.woff2 200 15.563 ms - -
127.0.0.1 - [19/Mar/2017:02:09:52 +0000] "GET /admin/_filemanagerinternal/css/roboto/medium-cyrillic-ext.woff2 HTTP/1.1" 200 15292
127.0.0.1 - [19/Mar/2017:02:09:52 +0000] "GET /admin/_filemanagerinternal/css/roboto/medium-cyrillic.woff2 HTTP/1.1" 200 10060
GET /admin/_filemanagerinternal/css/roboto/medium-cyrillic-ext.woff2 200 6.466 ms - -
GET /admin/_filemanagerinternal/css/roboto/medium-cyrillic.woff2 200 6.469 ms - -
127.0.0.1 - [19/Mar/2017:02:09:52 +0000] "GET /admin/_filemanagerinternal/css/roboto/medium-greek-ext.woff2 HTTP/1.1" 200 1412
GET /admin/_filemanagerinternal/css/roboto/medium-greek-ext.woff2 200 6.094 ms - 1412
127.0.0.1 - [19/Mar/2017:02:09:52 +0000] "GET /admin/_filemanagerinternal/css/roboto/medium-greek.woff2 HTTP/1.1" 200 7176
GET /admin/_filemanagerinternal/css/roboto/medium-greek.woff2 200 1.553 ms - -
127.0.0.1 - [19/Mar/2017:02:09:52 +0000] "GET /admin/_filemanagerinternal/css/roboto/medium-vietnamese.woff2 HTTP/1.1" 200 4780
GET /admin/_filemanagerinternal/css/roboto/medium-vietnamese.woff2 200 1.198 ms - -
127.0.0.1 - [19/Mar/2017:02:09:52 +0000] "GET /admin/_filemanagerinternal/css/roboto/medium-latin-ext.woff2 HTTP/1.1" 200 11912
GET /admin/_filemanagerinternal/css/roboto/medium-latin-ext.woff2 200 3.455 ms - -
127.0.0.1 - [19/Mar/2017:02:09:52 +0000] "GET /admin/_filemanagerinternal/css/material/icons.woff2 HTTP/1.1" 200 44300
127.0.0.1 - [19/Mar/2017:02:09:52 +0000] "GET /admin/_filemanagerinternal/css/roboto/medium-latin.woff2 HTTP/1.1" 200 14596
GET /admin/_filemanagerinternal/css/material/icons.woff2 200 9.990 ms - -
GET /admin/_filemanagerinternal/css/roboto/medium-latin.woff2 200 10.508 ms - -
127.0.0.1 - [19/Mar/2017:02:09:55 +0000] "GET /admin/config.toml HTTP/1.1" 200 23829
GET /admin/config.toml 200 51.812 ms - -
127.0.0.1 - [19/Mar/2017:02:09:55 +0000] "GET /admin/_filemanagerinternal/css/fonts.css HTTP/1.1" 200 4215
127.0.0.1 - [19/Mar/2017:02:09:55 +0000] "GET /admin/_filemanagerinternal/css/styles.css HTTP/1.1" 200 18865
127.0.0.1 - [19/Mar/2017:02:09:55 +0000] "GET /admin/_filemanagerinternal/js/vendor/form2js.js HTTP/1.1" 200 10092
127.0.0.1 - [19/Mar/2017:02:09:55 +0000] "GET /admin/_filemanagerinternal/js/common.js HTTP/1.1" 200 14429
127.0.0.1 - [19/Mar/2017:02:09:55 +0000] "GET /admin/_filemanagerinternal/css/normalize.css HTTP/1.1" 200 8053
GET /admin/_filemanagerinternal/css/fonts.css 200 7.171 ms - -
GET /admin/_filemanagerinternal/css/styles.css 200 8.712 ms - -
GET /admin/_filemanagerinternal/js/vendor/form2js.js 200 10.487 ms - -
GET /admin/_filemanagerinternal/js/common.js 200 11.467 ms - -
GET /admin/_filemanagerinternal/css/normalize.css 200 12.469 ms - -
127.0.0.1 - [19/Mar/2017:02:09:55 +0000] "GET /admin/_filemanagerinternal/js/vendor/ace/ace.js HTTP/1.1" 200 354518
GET /admin/_filemanagerinternal/js/vendor/ace/ace.js 200 25.941 ms - -
127.0.0.1 - [19/Mar/2017:02:09:55 +0000] "GET /admin/_filemanagerinternal/js/editor.js HTTP/1.1" 200 8500
GET /admin/_filemanagerinternal/js/editor.js 200 3.488 ms - -
127.0.0.1 - [19/Mar/2017:02:09:55 +0000] "GET /admin/_hugointernal/js/application.js HTTP/1.1" 200 4429
GET /admin/_hugointernal/js/application.js 200 1.605 ms - -
127.0.0.1 - [19/Mar/2017:02:09:55 +0000] "GET /admin/_filemanagerinternal/css/roboto/normal-greek.woff2 HTTP/1.1" 200 7208
127.0.0.1 - [19/Mar/2017:02:09:55 +0000] "GET /admin/_filemanagerinternal/css/roboto/normal-greek-ext.woff2 HTTP/1.1" 200 1432
127.0.0.1 - [19/Mar/2017:02:09:55 +0000] "GET /admin/_filemanagerinternal/css/roboto/normal-cyrillic-ext.woff2 HTTP/1.1" 200 15832
GET /admin/_filemanagerinternal/css/roboto/normal-greek.woff2 200 5.092 ms - -
GET /admin/_filemanagerinternal/css/roboto/normal-greek-ext.woff2 200 5.547 ms - 1432
GET /admin/_filemanagerinternal/css/roboto/normal-cyrillic-ext.woff2 200 6.611 ms - -
127.0.0.1 - [19/Mar/2017:02:09:55 +0000] "GET /admin/_filemanagerinternal/css/roboto/normal-cyrillic.woff2 HTTP/1.1" 200 9996
127.0.0.1 - [19/Mar/2017:02:09:55 +0000] "GET /admin/_filemanagerinternal/css/roboto/normal-vietnamese.woff2 HTTP/1.1" 200 4896
GET /admin/_filemanagerinternal/css/roboto/normal-cyrillic.woff2 200 6.628 ms - -
GET /admin/_filemanagerinternal/css/roboto/normal-vietnamese.woff2 200 6.920 ms - -
127.0.0.1 - [19/Mar/2017:02:09:55 +0000] "GET /admin/_filemanagerinternal/css/roboto/normal-latin-ext.woff2 HTTP/1.1" 200 12008
127.0.0.1 - [19/Mar/2017:02:09:55 +0000] "GET /admin/_filemanagerinternal/css/roboto/normal-latin.woff2 HTTP/1.1" 200 14584
GET /admin/_filemanagerinternal/css/roboto/normal-latin-ext.woff2 200 2.341 ms - -
GET /admin/_filemanagerinternal/css/roboto/normal-latin.woff2 200 3.199 ms - -
127.0.0.1 - [19/Mar/2017:02:09:55 +0000] "GET /admin/_filemanagerinternal/css/roboto/medium-cyrillic-ext.woff2 HTTP/1.1" 200 15292
GET /admin/_filemanagerinternal/css/roboto/medium-cyrillic-ext.woff2 200 1.253 ms - -
127.0.0.1 - [19/Mar/2017:02:09:55 +0000] "GET /admin/_filemanagerinternal/css/roboto/medium-cyrillic.woff2 HTTP/1.1" 200 10060
GET /admin/_filemanagerinternal/css/roboto/medium-cyrillic.woff2 200 1.562 ms - -
127.0.0.1 - [19/Mar/2017:02:09:55 +0000] "GET /admin/_filemanagerinternal/css/roboto/medium-greek-ext.woff2 HTTP/1.1" 200 1412
GET /admin/_filemanagerinternal/css/roboto/medium-greek-ext.woff2 200 1.701 ms - 1412
127.0.0.1 - [19/Mar/2017:02:09:55 +0000] "GET /admin/_filemanagerinternal/css/roboto/medium-greek.woff2 HTTP/1.1" 200 7176
GET /admin/_filemanagerinternal/css/roboto/medium-greek.woff2 200 2.154 ms - -
127.0.0.1 - [19/Mar/2017:02:09:55 +0000] "GET /admin/_filemanagerinternal/css/roboto/medium-vietnamese.woff2 HTTP/1.1" 200 4780
127.0.0.1 - [19/Mar/2017:02:09:55 +0000] "GET /admin/_filemanagerinternal/css/material/icons.woff2 HTTP/1.1" 200 44300
127.0.0.1 - [19/Mar/2017:02:09:55 +0000] "GET /admin/_filemanagerinternal/css/roboto/medium-latin-ext.woff2 HTTP/1.1" 200 11912
127.0.0.1 - [19/Mar/2017:02:09:55 +0000] "GET /admin/_filemanagerinternal/css/roboto/medium-latin.woff2 HTTP/1.1" 200 14596
GET /admin/_filemanagerinternal/css/roboto/medium-vietnamese.woff2 200 5.447 ms - -
GET /admin/_filemanagerinternal/css/roboto/medium-latin.woff2 200 5.744 ms - -
GET /admin/_filemanagerinternal/css/material/icons.woff2 200 7.045 ms - -
GET /admin/_filemanagerinternal/css/roboto/medium-latin-ext.woff2 200 8.241 ms - -
[master 1e1a19c] Saved /var/git/config.toml via administrative interface.
 1 file changed, 1 insertion(+), 42 deletions(-)
 rewrite config.toml (100%)
HEAD is now at 1e1a19c Saved /var/git/config.toml via administrative interface.
Error: While parsing config: (1, 1): unexpected token
19/Mar/2017:02:10:23 +0000 [ERROR 500 /admin/webdav/config.toml] exit status 255
127.0.0.1 - [19/Mar/2017:02:10:23 +0000] "PUT /admin/webdav/config.toml HTTP/1.1" 500 33
PUT /admin/webdav/config.toml 201 302.756 ms - 33

```

**Comment by ndarilek** [comment]
Do you have any info on how to build from source?

I know this is normally easy with Go projects, but what I'm *really* 
after is the Hugo plugin, which uses this plugin, which runs inside of 
Caddy. So how do I get a Caddy build that includes both these plugins? 
These are the only two I need for this project. I don't know if 
requiring the Hugo plugin also handles building this one or not.

Thanks.

**Comment by hacdias** [comment]
Hey!

@ndarilek I've the instructions here (https://github.com/hacdias/caddy-filemanager/wiki#developers). If you have any doubt, tell me 😄

**Comment by ndarilek** [comment]
Thanks. This issue is filed against filemanager, but it occurs using the 
Hugo plugin. I'm assuming the instructions are similar for that plugin too?

And, if so, do I need to `go get` this one, or is it enough to just `go 
get` and import the Hugo plugin?

Thanks.

**Comment by hacdias** [comment]
Yup! don't forget `go generate` in both.

**Comment by hacdias** [comment]
Closing due to inactivity. Although, if I'm right, this issue was fixed between revision 02e2bec and the most recent one. 😄
